torch/fx/passes/infra/partitioner.py

```diff
@@ -141,38 +143,48 @@ def dfs_iter_find_cycle(all_user_nodes: set[Node]):
 
                 return False
 
-            # check if merge would create cyclic dependency.
+            # find new partition users if merge.
             all_user_nodes = set()
-            for node in merged_nodes:
-                for user_node in node.users:
-                    if user_node not in merged_nodes:
-                        all_user_nodes.add(user_node)
+            for self_user in partition_users[self_id]:
+                if self_user not in other_nodes:
+                    all_user_nodes.add(self_user)
+            for other_user in partition_users[other_id]:
+                if other_user not in self_nodes:
+                    all_user_nodes.add(other_user)
```

Comment:
```suggestion
            all_user_nodes = (partition_users[self_id] - other_nodes)
            all_user_nodes |= (partition_users[other_id] - self_nodes)
```
Wouldn't this set operation be more concise, faster, and more efficient?

Reply:
I think you are right. Thanks for your suggestions!